### ISSUE TITLE
HTTP client

### ISSUE THREAD

**Issue by lippling** [GaloisInc/haskell-tor#25]
Is there a way to integrate into existing  HTTP client libraries? Socket usage for HTTP requests is a bit inconvenient :-).

**Comment by acw** [comment]
Yes ... partially. You could use the [HTTP library](https://hackage.haskell.org/package/HTTP), and hook up a TCP stream via the [Network.Stream](https://hackage.haskell.org/package/HTTP-4000.3.3/docs/Network-Stream.html) interface. You'll still need to manually set up the connection, but you can give the rest of the interaction over to the HTTP library.
